src/main/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandler.java

```diff
@@ -4,20 +4,43 @@
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
 import uk.gov.hmcts.reform.civil.event.EvidenceUploadNotificationEvent;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.notification.EvidenceUploadApplicantNotificationHandler;
+import uk.gov.hmcts.reform.civil.notification.EvidenceUploadRespondentNotificationHandler;
 import uk.gov.hmcts.reform.civil.service.CoreCaseDataService;
 
-import static uk.gov.hmcts.reform.civil.callback.CaseEvent.EVIDENCE_UPLOAD_NOTIFICATION;
-
 @Slf4j
 @Service
 @RequiredArgsConstructor
 public class EvidenceUploadNotificationEventHandler {
 
     private final CoreCaseDataService coreCaseDataService;
+    private final CaseDetailsConverter caseDetailsConverter;
+
+    private final EvidenceUploadApplicantNotificationHandler evidenceUploadApplicantNotificationHandler;
+    private final EvidenceUploadRespondentNotificationHandler evidenceUploadRespondentNotificationHandler;
 
     @EventListener
     public void sendEvidenceUploadNotification(EvidenceUploadNotificationEvent event) {
-        coreCaseDataService.triggerEvent(event.getCaseId(), EVIDENCE_UPLOAD_NOTIFICATION);
+        CaseDetails caseDetails = coreCaseDataService.getCase(event.getCaseId());
+        CaseData caseData = caseDetailsConverter.toCaseData(caseDetails);
+        try {
+            evidenceUploadApplicantNotificationHandler.notifyApplicantEvidenceUpload(caseData);
+        } catch (Exception e) {
+            log.error("Failed to send email notification to applicant for case '{}'", event.getCaseId());
```

Comment:
Add a comment in this method or in each of the "catch" clauses explaining why it is acceptable that we log but don't rethrow. Something like "these email notifications are not critical to the business and provided as a courtesy, as the user can log in and see the new uploads regardless. Hence we treat any failure here just as a warning"

Also: if it's not critical that emails are sent, let's not pollute the "errors" log. Use `log.warn` instead.

Reply:
done